File: 1813-maximum-erasure-value/maximum-erasure-value.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        seen = {}
        res = 0
        curr = 0
        L = 0

        for i, n in enumerate(nums):
            if n in seen:
                idx = seen[n]
                while L <= idx:
                    curr -= nums[L]
                    L += 1
            seen[n] = i
            curr += n
            res = max(res, curr)
        return res

File: 1813-maximum-erasure-value/maximum-erasure-value.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        unordered_map<int, int> mp;
        int res = 0, L = 0;
        for (int R = 0; R < nums.size(); ++R) {
            mp[nums[R]]++;
            while (mp[nums[R]] > 1) {
                mp[nums[L]]--;
                if (!mp[nums[L]]) mp.erase(nums[L]);
                L++;
            }

            int sum = 0;
            for (const auto& [n, count] : mp) {
                sum += n;
            }

            res = max(res, sum);
        }
        return res;
    }
};